osu.Game/Screens/Menu/ButtonSystem.cs

```diff
@@ -274,6 +274,10 @@ private bool onOsuLogo()
                 case ButtonSystemState.Play:
                     buttonsPlay.First().Click();
                     return false;
+
+                // no sound should be played if the logo is clicked on while transitioning to song select
+                case ButtonSystemState.EnteringMode:
+                    return false;
```

Comment:
I would've changed the default case to return false as I think `ButtonSystemState.Initial` is the only one that'll play sounds. `Exit` that is handled by the default case can potentially play sounds if the logo can still be clicked at that state (shouldn't though).

This can also be simplified by removing the default case and returning false outside and breaking every case instead of `Initial`.